tests/plugins/widget/contextmenu.js

```diff
@@ -22,6 +22,56 @@
 	var getWidgetById = widgetTestsTools.getWidgetById;
 
 	bender.test( {
+		// #1901
+		'test SHIFT + F10 shortcut upon widget focus': function() {
+			var editor = this.editor;
+
+			editor.widgets.add( 'testevent', {
+				editables: {
+					foo: '.foo'
+				}
+			} );
+
+			this.editorBot.setData(
+				'<p id="p1">foo</p><div data-widget="testevent" id="w1"><p class="foo">foo</p></div>',
+				function() {
+					var w1 = getWidgetById( editor, 'w1' ),
+						opened = 0;
+					w1.on( 'contextMenu', function() {
```

Comment:
Manual check if the function has been called is fine, although it could be simplified using `sinon.spy`. Lets make use of this library here to simplify a bit the code. Something like:

```javascript
var spy = sinon.spy();
widget.on( 'contextMenu', spy );
// ... fire event etc
assert.isTrue( spy.calledOnce );
```

Reply:
Thanks for letting me know of this library, of course I used it as you suggested.